core/src/main/java/org/apache/accumulo/core/clientImpl/ConditionalWriterImpl.java

```diff
@@ -801,6 +831,8 @@ public Result write(ConditionalMutation mutation) {
   @Override
   public void close() {
     threadPool.shutdownNow();
+    threadPoolCleanable.clean(); // deregister the cleaner, will not call shutdownNow() because
+                                 // closed is now true
```

Comment:
The comment does not appear to be correct. The cleaner will not call shutdownNow because the closed is now true, it will not call shutdownNow because it was passed a NOOP runnable. It's not passed any kind of Closeable to be able to detect whether closed is true now or not. It's doing nothing solely because it was only given a NOOP runnable.